frame/stake-tracker/src/lib.rs

```diff
@@ -0,0 +1,208 @@
+// This file is part of Substrate.
+
+// Copyright (C) 2023 Parity Technologies (UK) Ltd.
+// SPDX-License-Identifier: Apache-2.0
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// 	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//! # Stake Tracker Pallet
+//!
+//! The Stake Tracker pallet is used to maintain a sorted list of voters by listening to the events
```

Comment:
This pallet does not know that it is tracking "voters". It is just tracking some "account ids" based on "score".

Reply:
Once we use this pallet for approval voter tracking, this doc is already outdated.

Reply:
> Once we use this pallet for approval voter tracking, this doc is already outdated.

I'm aware of that, as I did not want to add `TargetList` to the docs before the logic for it has been implemented. let's make this doc generic. My concern here is that we are documenting this as a "generic" pallet, when it's actually tightly coupled with Staking logic. This is the sort of thing that makes things really confusing for people trying to figure out what the pallet is actually for. 

At the very least this pallet knows that it's tracking two lists (when the next PR is done), because this is what is being passed to it along with the score types and we also use a bunch of logic that directly refers to either balance or vote score, using StakingInterface to operate on those. 

Just leaving this here should anyone find this useful while trying to grasp the purpose of this pallet.